internal/engine/fake_notify_test.go

```diff
@@ -0,0 +1,124 @@
+package engine
+
+import (
+	"sync"
+
+	"github.com/windmilleng/tilt/internal/watch"
+)
+
+type fakeMetaWatcher struct {
+	events chan watch.FileEvent
+	errors chan error
+
+	mu         sync.Mutex
+	subs       []chan watch.FileEvent
+	subsErrors []chan error
+}
+
+func newFakeMetaWatcher() *fakeMetaWatcher {
+	r := &fakeMetaWatcher{events: make(chan watch.FileEvent), errors: make(chan error)}
+	go r.loop()
+	return r
+}
+
+func (w *fakeMetaWatcher) newSub() (watch.Notify, error) {
+	subCh := make(chan watch.FileEvent)
+	errorCh := make(chan error)
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	w.subs = append(w.subs, subCh)
+	w.subsErrors = append(w.subsErrors, errorCh)
+	return newFakeWatcher(subCh, errorCh), nil
+}
+
+func (w *fakeMetaWatcher) getSubs() []chan watch.FileEvent {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	return w.subs
+}
+
+func (w *fakeMetaWatcher) getSubErrors() []chan error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	return w.subsErrors
+}
+
+func (w *fakeMetaWatcher) loop() {
+	for {
+		select {
+		case e, ok := <-w.events:
+			if !ok {
+				for _, sub := range w.getSubs() {
```

Comment:
probably more correct to put this in a defer at the top of the function? bc you want to close this in the other `return` case as well